### PR TITLE
justinytchen hybridcompute microsoft.hybrid compute 2024 07 31 preview

### PR DESCRIPTION
**Pull request by justinytchen**
# ARM (Control Plane) API Specification Update Pull Request 

> [!TIP]
> Overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

spec_pr_review_workflow_diagram

## Purpose of this PR

What's the purpose of this PR? Check the specific option that applies. This is **mandatory**!

  - [ ] New resource provider.
  - [x] New API version for an existing resource provider. (If API spec is not defined in TypeSpec, the PR should have been created in adherence to [OpenAPI specs PR creation guidance](https://aka.ms/azsdkdocs/createopenapispec)).
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix OpenAPI spec quality issues in S360.
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [x] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [x] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://aka.ms/azurerpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to the diagram Step 2, "ARM API changes review", for this PR.

## Additional information

<details>
<summary> Viewing API changes</summary>

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

</details>
<details>
<summary>Suppressing failures</summary>

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[suppressions guide](https://aka.ms/azsdk/pr-suppressions) to get approval.

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom. Please fill out the `Purpose of this PR` and `Due diligence checklist`.
- If you don't have permissions to remove or add labels to the PR, request `write access` per [aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories](https://aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories")
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For help with ARM review (PR workflow diagram Step 2), see https://aka.ms/azsdk/pr-arm-review.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

### CODE REVIEW

**Comment by justinytchen** [comment]
azp /run

**Comment by TimLovellSmith** [comment]
@justinytchen you shouldn't ordinarily remove those labels yourself, breaking change reviewers should add their approval. Do you consider the breaking change output has false positives, or what?

**Comment by justinytchen** [comment]
@TimLovellSmith Yeah, I'm a little confused why the changes are being marked as breaking, since relative to the previous preview version, I am only adding properties. I believed we did something similar for the previous preview API version PRs. For instance, here: https://github.com/Azure/azure-rest-api-specs/pull/28787

**Comment by TimLovellSmith** [comment]
>         "disks": {

this is a bugfix I assume...

---
Refers to: specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2024-07-31-preview/HybridCompute.json:3057 in 2d67019. [](commit_id = 2d670192bb5b2fcab1ad32aecd1996aed6245eca, deletion_comment = False)

**Comment by TimLovellSmith** [comment]
> @TimLovellSmith Yeah, I'm a little confused why the changes are being marked as breaking, since relative to the previous preview version, I am only adding properties. I believed we did something similar (removing the BreakingChange label) for the previous preview API version PRs because we faced the same issue. For instance, here: #28787

Breaking changes compared with the last *stable* version as well as the latest preview.

**Comment by TimLovellSmith** [comment]
>         "resources": {

Hmm I didn't think you're supposed to have "resources" as a top level property, is it not actually inside of MachineProperties?

---
Refers to: specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2024-07-31-preview/HybridCompute.json:3404 in 546e8d0. [](commit_id = 546e8d0b0c1faf2dc74288dd83d54c6afdeaa5cc, deletion_comment = False)

**Comment by TimLovellSmith** [comment]
>         "resources": {

Is this new in the latest preview version, or was it already in the previous stable version?

---
In reply to: [2211564674](https://github.com/Azure/azure-rest-api-specs/pull/29560#issuecomment-2211564674) [](http://example.com/codeflow?ancestors=2211564674)

---
Refers to: specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2024-07-31-preview/HybridCompute.json:3404 in 546e8d0. [](commit_id = 546e8d0b0c1faf2dc74288dd83d54c6afdeaa5cc, deletion_comment = False)

**Comment by TimLovellSmith** [comment]
>           "description": "The correlation ID passed in from RSM per upgrade."

is this readonly?

---
Refers to: specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2024-07-31-preview/HybridCompute.json:3114 in 546e8d0. [](commit_id = 546e8d0b0c1faf2dc74288dd83d54c6afdeaa5cc, deletion_comment = False)

**Comment by TimLovellSmith** [comment]
>           "description": "The correlation ID passed in from RSM per upgrade."

Are docs about RSM going to make sense to customers?

---
Refers to: specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2024-07-31-preview/HybridCompute.json:3114 in 546e8d0. [](commit_id = 546e8d0b0c1faf2dc74288dd83d54c6afdeaa5cc, deletion_comment = False)

File: specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2024-07-31-preview/privateLinkScopes.json
Comment (TimLovellSmith) [inline specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2024-07-31-preview/privateLinkScopes.json:867]: "$ref": "../../../../../common-types/resource-management/v5/networksecuritypreimeter.json#/definitions/NetworkSecurityPerimeterConfiguration

File: specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2024-07-31-preview/privateLinkScopes.json
Comment (TimLovellSmith) [inline specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2024-07-31-preview/privateLinkScopes.json:909]: >"$ref": "#/definitions/NetworkSecurityPerimeterConfigurationListResult" [](http://example.com/codeflow?start=14&length=71)

"$ref": "../../../../../common-types/resource-management/v5/networksecuritypreimeter.json#/definitions/NetworkSecurityPerimeterConfigurationListResult"

File: specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2024-07-31-preview/privateLinkScopes.json
Comment (TimLovellSmith) [inline specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2024-07-31-preview/privateLinkScopes.json:1085]: >NetworkSecurityPerimeterConfiguration [](http://example.com/codeflow?start=5&length=37)

hopefully redundant with common-types?

File: specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2024-07-31-preview/privateLinkScopes.json
Comment (TimLovellSmith) [inline specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2024-07-31-preview/privateLinkScopes.json:1112]: >NetworkSecurityPerimeterConfigurationProperties [](http://example.com/codeflow?start=5&length=47)

hopefully redundant with common-types?

File: specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2024-07-31-preview/privateLinkScopes.json
Comment (TimLovellSmith) [inline specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2024-07-31-preview/privateLinkScopes.json:1144]: >NetworkSecurityPerimeterProfile [](http://example.com/codeflow?start=5&length=31)

hopefully redundant with common-types?

File: specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2024-07-31-preview/privateLinkScopes.json
Comment (TimLovellSmith) [inline specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2024-07-31-preview/privateLinkScopes.json:1184]: >ProvisioningIssue [](http://example.com/codeflow?start=5&length=17)

hopefully redundant with common-types?

File: specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2024-07-31-preview/privateLinkScopes.json
Comment (TimLovellSmith) [inline specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2024-07-31-preview/privateLinkScopes.json:1201]: >ProvisioningIssueProperties [](http://example.com/codeflow?start=5&length=27)

hopefully redundant with common-types?

File: specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2024-07-31-preview/privateLinkScopes.json
Comment (TimLovellSmith) [inline specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2024-07-31-preview/privateLinkScopes.json:1240]: >ProvisioningIssueType [](http://example.com/codeflow?start=5&length=21)

hopefully redundant with common-types?

File: specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2024-07-31-preview/privateLinkScopes.json
Comment (TimLovellSmith) [inline specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2024-07-31-preview/privateLinkScopes.json:1272]: >ProvisioningIssueSeverity [](http://example.com/codeflow?start=5&length=25)

hopefully redundant with common-types?

File: specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2024-07-31-preview/privateLinkScopes.json
Comment (TimLovellSmith) [inline specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2024-07-31-preview/privateLinkScopes.json:1294]: >AccessMode [](http://example.com/codeflow?start=5&length=10)

hopefully redundant with common-types?

File: specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2024-07-31-preview/privateLinkScopes.json
Comment (TimLovellSmith) [inline specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2024-07-31-preview/privateLinkScopes.json:1321]: >AccessRuleDirection [](http://example.com/codeflow?start=5&length=19)

hopefully redundant with common-types?

File: specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2024-07-31-preview/privateLinkScopes.json
Comment (TimLovellSmith) [inline specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2024-07-31-preview/privateLinkScopes.json:1343]: >AccessRule [](http://example.com/codeflow?start=5&length=10)

hopefully redundant with common-types?

File: specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2024-07-31-preview/privateLinkScopes.json
Comment (TimLovellSmith) [inline specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2024-07-31-preview/privateLinkScopes.json:1360]: >AccessRuleProperties [](http://example.com/codeflow?start=5&length=20)

hopefully redundant with common-types?

File: specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2024-07-31-preview/privateLinkScopes.json
Comment (TimLovellSmith) [inline specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2024-07-31-preview/privateLinkScopes.json:1379]: >NetworkSecurityPerimeter [](http://example.com/codeflow?start=5&length=24)

hopefully redundant with common-types?

File: specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2024-07-31-preview/privateLinkScopes.json
Comment (TimLovellSmith) [inline specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2024-07-31-preview/privateLinkScopes.json:1400]: >NetworkSecurityPerimeterConfigurationListResult [](http://example.com/codeflow?start=5&length=47)

hopefully redundant with common-types?

File: specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2024-07-31-preview/privateLinkScopes.json
Comment (TimLovellSmith) [inline specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2024-07-31-preview/privateLinkScopes.json:1419]: >ResourceAssociation [](http://example.com/codeflow?start=5&length=19)

hopefully redundant with common-types?

File: specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2024-07-31-preview/privateLinkScopes.json
Comment (TimLovellSmith) [inline specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2024-07-31-preview/privateLinkScopes.json:433]: >PrivateEndpointConnection [](http://example.com/codeflow?start=37&length=25)

"$ref": "../../../../../common-types/resource-management/v5/privatelinks.json#/definitions//PrivateEndpointConnection"

File: specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2024-07-31-preview/privateLinkScopes.json
Comment (TimLovellSmith) [inline specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2024-07-31-preview/privateLinkScopes.json:1758]: >"PrivateEndpointConnection" [](http://example.com/codeflow?start=4&length=27)

Hopefully redundant with common types

File: specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2024-07-31-preview/privateLinkScopes.json
Comment (TimLovellSmith) [inline specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2024-07-31-preview/privateLinkScopes.json:1778]: >PrivateEndpointConnectionListResult [](http://example.com/codeflow?start=5&length=35)

Hopefully redundant with common types

**Review by TimLovellSmith — DISMISSED** [review]
@justinytchen So, by the way.... I just noticed you have networkSecurityPerimeter type definitions in here. 

I've been getting RPs to try and standardize on using common-types for that. Could you please give that a try, and we can check if there is any blocker to standardizing, like it would lead to any breaking changes?

https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v5/networksecurityperimeter.json

To be clear, I only want us to do this if its *non-breaking*, otherwise the benefits are unclear -- but I think in general if its non-breaking then there are clear maintainability benefits to getting it done.

**Comment by TimLovellSmith** [comment]
> > ```
> >     "resources": {
> > ```
> 
> Hmm I didn't think you're supposed to have "resources" as a top level property, is it not actually inside of MachineProperties?
> 
> Refers to: specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2024-07-31-preview/HybridCompute.json:3404 in [546e8d0](https://github.com/Azure/azure-rest-api-specs/commit/546e8d0b0c1faf2dc74288dd83d54c6afdeaa5cc). [](commit_id = [546e8d0](https://github.com/Azure/azure-rest-api-specs/commit/546e8d0b0c1faf2dc74288dd83d54c6afdeaa5cc), deletion_comment = False)

Could you double check that your service actually returns this as a top level property?
If you do, we don't need to panic and fix it right now, since that would, sigh, be breaking.

But if its actually a documentation bug of some kind we should fix it.

**Comment by justinytchen** [comment]
> > ```
> >     "resources": {
> > ```
> 
> Hmm I didn't think you're supposed to have "resources" as a top level property, is it not actually inside of MachineProperties?
> 
> Could you double check that your service actually returns this as a top level property? If you do, we don't need to panic and fix it right now, since that would, sigh, be breaking.
> 
> But if its actually a documentation bug of some kind we should fix it.
> 
> Refers to: specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2024-07-31-preview/HybridCompute.json:3404 in [546e8d0](https://github.com/Azure/azure-rest-api-specs/commit/546e8d0b0c1faf2dc74288dd83d54c6afdeaa5cc). [](commit_id = [546e8d0](https://github.com/Azure/azure-rest-api-specs/commit/546e8d0b0c1faf2dc74288dd83d54c6afdeaa5cc), deletion_comment = False)

We do return it as a top level property

**Comment by justinytchen** [comment]
> > ```
> >       "description": "The correlation ID passed in from RSM per upgrade."
> > ```
> 
> is this readonly?
> 
> Refers to: specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2024-07-31-preview/HybridCompute.json:3114 in [546e8d0](https://github.com/Azure/azure-rest-api-specs/commit/546e8d0b0c1faf2dc74288dd83d54c6afdeaa5cc). [](commit_id = [546e8d0](https://github.com/Azure/azure-rest-api-specs/commit/546e8d0b0c1faf2dc74288dd83d54c6afdeaa5cc), deletion_comment = False)

Not read-only - RSM will make PATCH calls to the machine and set the correlation ID, desired version, and other agent upgrade properties

**Comment by justinytchen** [comment]
> > ```
> >       "description": "The correlation ID passed in from RSM per upgrade."
> > ```
> 
> Are docs about RSM going to make sense to customers?
> 
> Refers to: specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2024-07-31-preview/HybridCompute.json:3114 in [546e8d0](https://github.com/Azure/azure-rest-api-specs/commit/546e8d0b0c1faf2dc74288dd83d54c6afdeaa5cc). [](commit_id = [546e8d0](https://github.com/Azure/azure-rest-api-specs/commit/546e8d0b0c1faf2dc74288dd83d54c6afdeaa5cc), deletion_comment = False)

Will fix to make it more understandable to customers

**Comment by justinytchen** [comment]
> @justinytchen So, by the way.... I just noticed you have networkSecurityPerimeter type definitions in here.
> 
> I've been getting RPs to try and standardize on using common-types for that. Could you please give that a try, and we can check if there is any blocker to standardizing, like it would lead to any breaking changes?
> 
> https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v5/networksecurityperimeter.json
> 
> To be clear, I only want us to do this if its _non-breaking_, otherwise the benefits are unclear -- but I think in general if its non-breaking then there are clear maintainability benefits to getting it done.

@TimLovellSmith I updated the references to use the definitions in common-types if it didn't cause a breaking change. 
Regarding the breaking changes relative to the latest stable, what are the next steps? 

We released a stable version back in 2022, and have released many preview versions since. One of those preview versions had a breaking change relative to the stable version, but the subsequent preview versions did not have a breaking change relative to that initial breaking change. 
Does this mean that every subsequent preview version we make now will have to go through a breaking change review

**Comment by madhura-np** [comment]
Use this link https://aka.ms/brch for breaking change process.

**Comment by raosuhas** [comment]
Please remove the ARMChangesRequested Label once you have got the breaking change approval

**Comment by mikekistler** [comment]
Breaking changes reviewed and approved previously in #29070

**Review by mikekistler — DISMISSED** [review]
Requesting changes on this PR to ensure it merges after #29831

File: specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2024-07-31-preview/HybridCompute.json
Comment (justinytchen) [inline specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2024-07-31-preview/HybridCompute.json:3078]: Why can this not be named "type"

File: specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2024-07-31-preview/HybridCompute.json
Comment (justinytchen) [inline specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2024-07-31-preview/HybridCompute.json:3087]: Why can this not be named "type"?

**Comment by TimLovellSmith** [comment]
@ jutinytchen can I hear what those were?

>Tried replacing, but there were too many breaking changes

**Comment by justinytchen** [comment]
@TimLovellSmith 
It was awhile back, it was mostly lots of small inconsistencies with properties and attributes like read-only, or slightly different enum values, types, etc

**Comment by justinytchen** [comment]
@mikekistler the stable api version has been approved and merged
https://github.com/Azure/azure-rest-api-specs/pull/29954/files

**Comment by justinytchen** [comment]
@mikekistler I'm a little confused as to why the pipelines are failing in the most recent build, but succeeded in the previous one, where the only change is whitespace change.

Currently failing: https://github.com/Azure/azure-rest-api-specs/runs/29128211035
Previously succeeding: https://github.com/Azure/azure-rest-api-specs/runs/28831105183
Diffs between the two commits: https://github.com/Azure/azure-rest-api-specs/pull/29560/commits/e583cccf66020dafca7dd8495ba003a4f2284f8e

**Comment by justinytchen** [comment]
@mikekistler I'm a little confused as to why the pipelines are failing in the most recent build, but succeeded in the previous one, where the only change is whitespace change.

Currently failing: https://github.com/Azure/azure-rest-api-specs/runs/29128211035
Previously succeeding: https://github.com/Azure/azure-rest-api-specs/runs/28831105183
Diffs between the two commits: https://github.com/Azure/azure-rest-api-specs/pull/29560/commits/e583cccf66020dafca7dd8495ba003a4f2284f8e

**Comment by justinytchen** [comment]
@razvanbadea-msft 
I'd like some guidance on how to address these new LIST API requirements that requires x-ms-pageable and disallows certain top-level properties like nextLink. It seems like this change was made relatively recently.
In this newly introduced API version as well as all our previous API versions, we use nextLink for list results and include certain top-level properties. I am unsure if we can modify our RP to remove those properties and add x-ms-pageable without breaking changes. Could we get an exception for these checks?

https://github.com/Azure/azure-rest-api-specs/pull/29560/checks?check_run_id=29130605294

**Comment by mentat9** [comment]
> @razvanbadea-msft I'd like some guidance on how to address these new LIST API requirements that requires x-ms-pageable and disallows certain top-level properties like nextLink. It seems like this change was made relatively recently. In this newly introduced API version as well as all our previous API versions, we use nextLink for list results and include other top-level properties. I am unsure if we can modify our RP to remove those properties and add x-ms-pageable without breaking changes. Could we get an exception for these checks?
> 
> https://github.com/Azure/azure-rest-api-specs/pull/29560/checks?check_run_id=29130605294

@justinytchen - The x-ms-pageable requirement has always been there, although I don't know when the linter checks added validation for it. The BodyTopLevelProperties errors against "value" and "nextLink" are a false positive due to a current bug in the check (the reviewer will bypass this check after the other errors are fixed).

However other BodyTopLevelProperties checks should be valid (this is a new-ish check). Other checks should also all be valid.

Checks on APIs that already shipped in stable APIs should be suppressed (look for "Suppressing failures" near the top of the PR). Checks on APIs being added or only previously published in preview should be fixed.

**Comment by justinytchen** [comment]
@mentat9 I've fixed one of the x-ms-pageable issues. The only ones that remain that are not the buggy top-level property issues are the following. 
| Rule             | Message                                                                             | Related RPC [For API reviewers] |
| ------------- | -------------------------------------------------------------- | ---------------------------------- |
|             ❌ [EvenSegmentedPathForPutOperation](https://github.com/Azure/azure-openapi-validator/blob/main/docs/even-segmented-path-for-put-operation.md)      | API path with PUT operation defined MUST have even number of segments (i.e. end in {resourceType}/{resourceName} segments). Location: [Microsoft.HybridCompute/preview/2024-07-31-preview/privateLinkScopes.json#L698](https://github.com/Azure/azure-rest-api-specs/blob/2770f062b4b2585dd05d57915d8ab1b54b70f8a7/specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2024-07-31-preview/privateLinkScopes.json#L698) |RPC-Put-V1-02|
|             ❌ [XmsPageableForListCalls](https://github.com/Azure/azure-openapi-validator/blob/main/docs/xms-pageable-for-list-calls.md)  | x-ms-pageable extension must be specified for LIST APIs. Location: [Microsoft.HybridCompute/preview/2024-07-31-preview/privateLinkScopes.json#L699](https://github.com/Azure/azure-rest-api-specs/blob/2770f062b4b2585dd05d57915d8ab1b54b70f8a7/specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2024-07-31-preview/privateLinkScopes.json#L699) |RPC-Get-V1-13|

For the first one, I believe this is a false-positive - we've had this API endpoint for the past few preview versions (though not stable)
For the second one I'm not sure why it's being flagged as a LIST API endpoint

**Comment by ramoka178** [comment]
@justinytchen  The lint issues are being flagged at "/{resourceUri}/providers/Microsoft.HybridCompute/networkConfigurations/current" . I see that this is only present in preview API versions and not in stable. 

Could you take this as an opportunity to fix this as this API path is an antipattern. 
https://github.com/Azure/azure-openapi-validator/blob/main/docs/even-segmented-path-for-put-operation.md

**Comment by ramoka178** [comment]
New path is also having issues. Also I dont think the RP name being given twice makes sense.

**Comment by justinytchen** [comment]
@ramoka178  I fixed the new path, though I'm not sure why it's still being flagged (I believe there are 12 segments).
The RP name given twice is intentional - it is an extension resource from the same provider

**Comment by ramoka178** [comment]
> @ramoka178 I fixed the new path, though I'm not sure why it's still being flagged (I believe there are 12 segments). The RP name given twice is intentional - it is an extension resource from the same provider. The example requests support this as well (see the previous API versions' examples)
> 
> https://github.com/Azure/azure-rest-api-specs/blob/12fadddfea11f0e4c40d8cd7ab038aa9383a7da3/specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2023-10-03-preview/examples/networkConfiguration/NetworkConfigurationsGet.json#L9

I am not sure about the example. Also, I have not seen the format of duplicating the RP name in the API itself.
I believe the API path should be of {resourceType}/{resourceName} format, which is not present for the last part of "providers/Microsoft.HybridCompute/networkConfigurations/current"

You could post your issue in https://teams.microsoft.com/l/channel/19%3A4f661242c446452e895359cc3ef45125%40thread.tacv2/ARM%20linter%20rules?groupId=0cab4ce9-7691-42ae-82e3-460d4346a710&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47 for any guidance.

Also there are other errors in [BodyTopLevelProperties](https://github.com/Azure/azure-openapi-validator/blob/main/docs/body-top-level-properties.md) , that are not related to value/nextLink issue. Please fix them as well.

**Comment by justinytchen** [comment]
Previous preview API version with AgentVersion top level properties: https://github.com/Azure/azure-rest-api-specs/pull/29070

**Comment by ramoka178** [comment]
> Went to office hours, talked about the networkConfiguration/current path, which would be discussed internally (in the ARM team) to figure out the resolution. Also discussed the AgentVersion response, where the top level properties has already been approved. Both of these changes have been in the past few preview API versions and are released, so it would be difficult to change (i.e. breaking change)

For next API reviewer: Asked this question on https://teams.microsoft.com/l/message/19:78cd6c72c41240dc84a36523af60b88a@thread.tacv2/1725044502273?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=0cab4ce9-7691-42ae-82e3-460d4346a710&parentMessageId=1725044502273&teamName=ARM%20API%20Reviewers&channelName=ARM%20swagger%20reviews&createdTime=1725044502273

**Review by vedkale — APPROVED** [review]

**Review by jiaqi-beep — APPROVED** [review]

**Pull request merged**
